From dcfda5527538993dc7b2291cb9b9b9967f3b3c4a Mon Sep 17 00:00:00 2001
From: David Sterba <dsterba@suse.com>
Date: Wed, 10 Feb 2021 16:36:50 +0100
Subject: [PATCH] btrfs-progs: fix device mapper path canonicalization

Commit 922eaa7b5472 ("btrfs-progs: build: fix linking with static
libmount") broke path canonicalization, that prevented eg 'device add
/dev/dm-0' to properly recognize the device mapper names.

Issue: #339
Signed-off-by: David Sterba <dsterba@suse.com>
---
 common/path-utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/path-utils.c b/common/path-utils.c
index d4809a6ae..e38513f68 100644
--- a/common/path-utils.c
+++ b/common/path-utils.c
@@ -325,7 +325,7 @@ char *path_canonicalize(const char *path)
 		return strdup(path);
 	p = strrchr(canonical, '/');
 	if (p && strncmp(p, "/dm-", 4) == 0 && isdigit(*(p + 4))) {
-		char *dm = path_canonicalize(p + 1);
+		char *dm = path_canonicalize_dm_name(p + 1);
 
 		if (dm) {
 			free(canonical);
